packages/types/src/base.ts

```diff
@@ -2,12 +2,15 @@ import type { AccountId, AssetId, ChainId } from '@shapeshiftoss/caip'
 
 import type { PartialRecord } from './utility'
 
-export type BIP44Params = {
+export type DefaultBIP44Params = {
   purpose: number
   coinType: number
   accountNumber: number
-  isChange?: boolean
-  index?: number
+}
+
+export type BIP44Params = DefaultBIP44Params & {
+  isChange: boolean | undefined
+  index: number | undefined
```

Comment:
@kaladinlight do we want to call this `address_index` so that we're explicitly honouring the BIP44 terminology here, or is that too early and probably a better fit for the actual custom derivation work?

Similarly, I believe `accountNumber` could be renamed `account_index` to match the BIP44 spec when we get there

Reply:
happy to update to `addressIndex`, but going to leave `accountNumber`. reason being, we use this throughout the codebase and our terminology in web is `accountNumber` vernacular, so changing would just confuse things for no substantial gain imo.

Reply:
Fine with that @kaladinlight! We may or may not come back to this when we support custom derivation paths where we increment the address index instead (and what we call an account number is an address_index increase) but agree for the time being would just add confusion.

Reply:
9e079dcac871fc638e895b9e82a05d2425c8d1bb